airbyte-integrations/connectors/source-google-ads/source_google_ads/config_migrations.py

```diff
@@ -0,0 +1,130 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+
+import logging
+from typing import Any, List, Mapping
+
+from airbyte_cdk.config_observation import create_connector_config_control_message
+from airbyte_cdk.entrypoint import AirbyteEntrypoint
+from airbyte_cdk.models import FailureType
+from airbyte_cdk.sources import Source
+from airbyte_cdk.sources.message import InMemoryMessageRepository, MessageRepository
+from airbyte_cdk.utils import AirbyteTracedException
+
+from .utils import GAQL
+
+logger = logging.getLogger("airbyte_logger")
+
+FULL_REFRESH_CUSTOM_TABLE = [
+    "asset",
+    "asset_group_listing_group_filter",
+    "custom_audience",
+    "geo_target_constant",
+    "change_event",
+    "change_status",
+]
+
+
+class MigrateCustomQuery:
+    """
+    This class stands for migrating the config at runtime,
+    while providing the backward compatibility when falling back to the previous source version.
```

Comment:
Could you please share how backward compatibility is guaranteed?

Reply:
I updated the docs. Backward compatibility is achieved by creating a new property instead of editing an existing one.